Kodi fails to build with libfmt 10.0.0 - Removed deprecated implicit conversions for enums and conversions to primitive types

<!--- Please fill out this template to the best of your ability. You can always edit this issue once you have created it. -->
<!--- Read the following link before you create a new problem report: https://kodi.wiki/view/HOW-TO:Submit_a_bug_report  -->
## Bug report
### Describe the bug
Here is a clear and concise description of what the problem is:
<!--- Provide a more detailed introduction to the issue itself, and why you consider it to be a bug -->
<!--- A bug report that is not clear will be closed -->
<!--- Put your text below this line -->
**libfmt 10.0.0 removes the enum conversions.**

a couple examples are:
- xbmc/filesystem/CurlFile.cpp
https://github.com/curl/curl/blob/891e25edb8527bb8de79cdca6d943216c230e905/include/curl/curl.h#L792

- xbmc/games/addons/streams/GameClientStreamVideo.cpp
  - CLog::Log(LOGERROR, "GAME: Unknown pixel format: {}", properties.format);
  - Where format is of type GAME_PIXEL_FORMAT https://github.com/xbmc/xbmc/blob/acca69baa2eae65123e78ee2f77249181725ef5d/xbmc/addons/kodi-dev-kit/include/kodi/c-api/addon-instance/game.h#L201

## Expected Behavior
Here is a clear and concise description of what was expected to happen:
<!--- Tell us what should happen -->
<!--- Put your text below this line -->
Details of change in libfmt at:
- https://github.com/fmtlib/fmt/releases/tag/10.0.0
- https://fmt.dev/latest/api.html#udt

## Actual Behavior
<!--- Tell us what happens instead -->
<!--- Put your text below this line -->
Compile fails
```
FAILED: build/filesystem/CMakeFiles/filesystem.dir/CurlFile.cpp.o 
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-12.0-devel/toolchain/bin/x86_64-libreelec-linux-gnu-g++ -DNPT_CONFIG_ENABLE_LOGGING -I/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-12.0-devel/build/kodi-21.0a2-Omega -I/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-12.0-devel/build/kodi-21.0a2-Omega/lib -I/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-12.0-devel/build/kodi-21.0a2-Omega/xbmc -I/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-12.0-devel/build/kodi-21.0a2-Omega/xbmc/platform/linux -I/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-12.0-devel/build/kodi-21.0a2-Omega/xbmc/cores/VideoPlayer -I/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-12.0-devel/build/kodi-21.0a2-Omega/.x86_64-libreelec-linux-gnu/build -I/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-12.0-devel/build/kodi-21.0a2-Omega/.x86_64-libreelec-linux-gnu/build/include -I/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-12.0-devel/build/kodi-21.0a2-Omega/xbmc/platform/posix -isystem /var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-12.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/dbus-1.0 -isystem /var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-12.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/lib/dbus-1.0/include -isystem /var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-12.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/python3.11 -isystem /var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-12.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/samba-4.0 -isystem /var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-12.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/libxml2 -isystem /var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-12.0-devel/build/kodi-21.0a2-Omega/.x86_64-libreelec-linux-gnu/build/cores/RetroPlayer/messages -isystem /var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-12.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/freetype2 -isystem /var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-12.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/fribidi -isystem /var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-12.0-devel/build/kodi-21.0a2-Omega/xbmc/contrib -isystem /var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-12.0-devel/build/kodi-21.0a2-Omega/depends/include -isystem /var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-12.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/lzo -isystem /var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-12.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/libdrm -isystem /var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-12.0-devel/build/kodi-21.0a2-Omega/lib/libUPnP -isystem /var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-12.0-devel/build/kodi-21.0a2-Omega/lib/libUPnP/Platinum/Source/Core -isystem /var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-12.0-devel/build/kodi-21.0a2-Omega/lib/libUPnP/Platinum/Source/Platinum -isystem /var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-12.0-devel/build/kodi-21.0a2-Omega/lib/libUPnP/Platinum/Source/Devices/MediaConnect -isystem /var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-12.0-devel/build/kodi-21.0a2-Omega/lib/libUPnP/Platinum/Source/Devices/MediaRenderer -isystem /var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-12.0-devel/build/kodi-21.0a2-Omega/lib/libUPnP/Platinum/Source/Devices/MediaServer -isystem /var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-12.0-devel/build/kodi-21.0a2-Omega/lib/libUPnP/Neptune/Source/Core -isystem /var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-12.0-devel/build/kodi-21.0a2-Omega/lib/libUPnP/Neptune/Source/System/Posix -march=x86-64-v3 -Wall -pipe  -O3 -fomit-frame-pointer -DNDEBUG -Wdouble-promotion -Wmissing-field-initializers -Wsign-compare -Wextra -Wno-unused-parameter -Wnon-virtual-dtor  -s -std=c++17 -flto=16 -fno-fat-lto-objects -DTARGET_POSIX -DTARGET_LINUX -D_GNU_SOURCE -DHAVE_LINUX_UDMABUF=1 -DHAVE_LINUX_DMA_HEAP=1 -DHAVE_LINUX_DMA_BUF=1 -DHAVE_MKOSTEMP=1 -DHAVE_LINUX_MEMFD=1 -DHAVE_STATX=1 -DHAVE_SSE=1 -DHAVE_SSE2=1 -DHAVE_SSE3=1 -DHAVE_SSSE3=1 -DHAVE_SSE4_1=1 -D__STDC_CONSTANT_MACROS -D_FILE_OFFSET_BITS=64 -DHAS_POSIX_NETWORK -DHAS_LINUX_NETWORK -DHAS_BUILTIN_SYNC_ADD_AND_FETCH=1 -DHAS_BUILTIN_SYNC_SUB_AND_FETCH=1 -DHAS_BUILTIN_SYNC_VAL_COMPARE_AND_SWAP=1 -DHAVE_INOTIFY=1 -DHAVE_POSIX_FADVISE=1 -DHAVE_LOCALTIME_R=1 -DHAVE_GMTIME_R=1 -DHAVE_INTTYPES_H=1 -DHAS_ALSA=1 -DHAS_AVAHI=1 -DHAS_ZEROCONF=1 -DHAVE_LIBCEC=1 -DHAS_DBUS=1 -DHAS_ISO9660PP=1 -DHAS_LIRC=1 -DHAS_WEB_SERVER=1 -DHAS_WEB_INTERFACE=1 -DHAS_FILESYSTEM_NFS=1 -DHAS_NFS_SET_TIMEOUT -DHAS_NFS_MOUNT_GETEXPORTS_TIMEOUT -DHAS_AIRPLAY=1 -DHAS_PULSEAUDIO=1 -DHAS_PYTHON=1 -DHAS_FILESYSTEM_SMB=1 -DHAVE_LIBUDEV=1 -DHAS_UDFREAD=1 -DHAVE_LIBXSLT=1 -DHAVE_LIBVA=1 -DHAVE_NEW_CROSSGUID -DFFMPEG_VER_SHA=\"6.0.0\" -DSPDLOG_FMT_EXTERNAL -DSPDLOG_DEBUG_ON -DSPDLOG_NO_ATOMIC_LEVELS -DSPDLOG_ENABLE_PATTERN_PADDING -DSPDLOG_COMPILED_LIB -DSPDLOG_SHARED_LIB -I/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-12.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include -DHAVE_GBM=1 -DHAS_GBM_BO_MAP=1 -DHAS_GBM_MODIFIERS=1 -DHAVE_HDR_OUTPUT_METADATA=1 -DHAVE_DRM_MODIFIER_NAME=1 -DHAS_GLES=3 -DHAS_EGL=1 -DHAVE_EGLEXTANGLE=1 -DHAS_MARIADB=1 -DHAS_UPNP=1 -DHAS_DVD_DRIVE -DHAS_CDDA_RIPPER -DHAS_AIRTUNES=1 -DBIN_INSTALL_PATH=\"/usr/lib/kodi\" -DINSTALL_PATH=\"/usr/share/kodi\" -Werror=double-promotion -Werror=missing-field-initializers -Werror=sign-compare -MD -MT build/filesystem/CMakeFiles/filesystem.dir/CurlFile.cpp.o -MF build/filesystem/CMakeFiles/filesystem.dir/CurlFile.cpp.o.d -o build/filesystem/CMakeFiles/filesystem.dir/CurlFile.cpp.o -c /var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-12.0-devel/build/kodi-21.0a2-Omega/xbmc/filesystem/CurlFile.cpp
In file included from /var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-12.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/spdlog/fmt/fmt.h:31,
                 from /var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-12.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/spdlog/common.h:45,
                 from /var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-12.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/spdlog/spdlog.h:12,
                 from ../xbmc/utils/log.h:36,
                 from ../xbmc/threads/SystemClock.h:11,
                 from ../xbmc/filesystem/CurlFile.cpp:19:
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-12.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/fmt/core.h: In instantiation of 'constexpr fmt::v10::detail::value<Context> fmt::v10::detail::make_value(T&&) [with Context = fmt::v10::basic_format_context<fmt::v10::appender, char>; T = CURLcode&]':
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-12.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/fmt/core.h:1711:29:   required from 'constexpr fmt::v10::detail::value<Context> fmt::v10::detail::make_arg(T&&) [with bool IS_PACKED = true; Context = fmt::v10::basic_format_context<fmt::v10::appender, char>; type <anonymous> = fmt::v10::detail::type::custom_type; T = CURLcode&; typename std::enable_if<IS_PACKED, int>::type <anonymous> = 0]'
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-12.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/fmt/core.h:1827:77:   required from 'constexpr fmt::v10::format_arg_store<Context, Args>::format_arg_store(T&& ...) [with T = {const char (&)[7], std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, const char*&, CURLcode&}; Context = fmt::v10::basic_format_context<fmt::v10::appender, char>; Args = {char [7], std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, const char*, CURLcode}]'
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-12.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/fmt/core.h:1844:31:   required from 'constexpr fmt::v10::format_arg_store<Context, typename std::remove_cv<typename std::remove_reference<T>::type>::type ...> fmt::v10::make_format_args(T&& ...) [with Context = basic_format_context<appender, char>; T = {const char (&)[7], std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, const char*&, CURLcode&}]'
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-12.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/fmt/core.h:2817:44:   required from 'std::string fmt::v10::format(format_string<T ...>, T&& ...) [with T = {const char (&)[7], std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, const char*, CURLcode&}; std::string = std::__cxx11::basic_string<char>; format_string<T ...> = basic_format_string<char, const char (&)[7], std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, const char*, CURLcode&>]'
../xbmc/utils/log.h:129:31:   required from 'void CLog::FormatAndLogInternal(spdlog::level::level_enum, const std::string_view&, Args&& ...) [with Args = {const char (&)[7], std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, const char*, CURLcode&}; std::string_view = std::basic_string_view<char>]'
../xbmc/utils/log.h:100:39:   required from 'static void CLog::Log(spdlog::level::level_enum, const std::string_view&, Args&& ...) [with Args = {const char (&)[7], std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, const char*, CURLcode&}; std::string_view = std::basic_string_view<char>]'
../xbmc/utils/log.h:80:8:   required from 'static void CLog::Log(int, const std::string_view&, Args&& ...) [with Args = {const char (&)[7], std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, const char*, CURLcode&}; std::string_view = std::basic_string_view<char>]'
../xbmc/filesystem/CurlFile.cpp:1381:14:   required from here
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-12.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/fmt/core.h:1691:7: error: static assertion failed: Cannot format an argument. To make type T formattable provide a formatter<T> specialization: https://fmt.dev/latest/api.html#udt
 1691 |       formattable,
      |       ^~~~~~~~~~~
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-12.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/fmt/core.h:1691:7: note: 'formattable' evaluates to false
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-12.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/fmt/core.h: In instantiation of 'constexpr fmt::v10::detail::value<Context> fmt::v10::detail::make_value(T&&) [with Context = fmt::v10::basic_format_context<fmt::v10::appender, char>; T = CURLMcode&]':
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-12.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/fmt/core.h:1711:29:   required from 'constexpr fmt::v10::detail::value<Context> fmt::v10::detail::make_arg(T&&) [with bool IS_PACKED = true; Context = fmt::v10::basic_format_context<fmt::v10::appender, char>; type <anonymous> = fmt::v10::detail::type::custom_type; T = CURLMcode&; typename std::enable_if<IS_PACKED, int>::type <anonymous> = 0]'
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-12.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/fmt/core.h:1827:77:   required from 'constexpr fmt::v10::format_arg_store<Context, Args>::format_arg_store(T&& ...) [with T = {const char (&)[11], const void*&, CURLMcode&}; Context = fmt::v10::basic_format_context<fmt::v10::appender, char>; Args = {char [11], const void*, CURLMcode}]'
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-12.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/fmt/core.h:1844:31:   required from 'constexpr fmt::v10::format_arg_store<Context, typename std::remove_cv<typename std::remove_reference<T>::type>::type ...> fmt::v10::make_format_args(T&& ...) [with Context = basic_format_context<appender, char>; T = {const char (&)[11], const void*&, CURLMcode&}]'
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-12.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/fmt/core.h:2817:44:   required from 'std::string fmt::v10::format(format_string<T ...>, T&& ...) [with T = {const char (&)[11], const void*, CURLMcode&}; std::string = std::__cxx11::basic_string<char>; format_string<T ...> = basic_format_string<char, const char (&)[11], const void*, CURLMcode&>]'
../xbmc/utils/log.h:129:31:   required from 'void CLog::FormatAndLogInternal(spdlog::level::level_enum, const std::string_view&, Args&& ...) [with Args = {const char (&)[11], const void*, CURLMcode&}; std::string_view = std::basic_string_view<char>]'
../xbmc/utils/log.h:100:39:   required from 'static void CLog::Log(spdlog::level::level_enum, const std::string_view&, Args&& ...) [with Args = {const char (&)[11], const void*, CURLMcode&}; std::string_view = std::basic_string_view<char>]'
../xbmc/utils/log.h:80:8:   required from 'static void CLog::Log(int, const std::string_view&, Args&& ...) [with Args = {const char (&)[11], const void*, CURLMcode&}; std::string_view = std::basic_string_view<char>]'
../xbmc/filesystem/CurlFile.cpp:1899:18:   required from here
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-12.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/fmt/core.h:1691:7: error: static assertion failed: Cannot format an argument. To make type T formattable provide a formatter<T> specialization: https://fmt.dev/latest/api.html#udt
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-12.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/fmt/core.h:1691:7: note: 'formattable' evaluates to false
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-12.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/fmt/core.h: In instantiation of 'constexpr fmt::v10::detail::value<Context> fmt::v10::detail::make_value(T&&) [with Context = fmt::v10::basic_format_context<fmt::v10::appender, char>; T = curl_proxytype&]':
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-12.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/fmt/core.h:1711:29:   required from 'constexpr fmt::v10::detail::value<Context> fmt::v10::detail::make_arg(T&&) [with bool IS_PACKED = true; Context = fmt::v10::basic_format_context<fmt::v10::appender, char>; type <anonymous> = fmt::v10::detail::type::custom_type; T = curl_proxytype&; typename std::enable_if<IS_PACKED, int>::type <anonymous> = 0]'
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-12.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/fmt/core.h:1827:77:   required from 'constexpr fmt::v10::format_arg_store<Context, Args>::format_arg_store(T&& ...) [with T = {const char (&)[19], std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, curl_proxytype&}; Context = fmt::v10::basic_format_context<fmt::v10::appender, char>; Args = {char [19], std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, curl_proxytype}]'
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-12.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/fmt/core.h:1844:31:   required from 'constexpr fmt::v10::format_arg_store<Context, typename std::remove_cv<typename std::remove_reference<T>::type>::type ...> fmt::v10::make_format_args(T&& ...) [with Context = basic_format_context<appender, char>; T = {const char (&)[19], std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, curl_proxytype&}]'
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-12.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/fmt/core.h:2817:44:   required from 'std::string fmt::v10::format(format_string<T ...>, T&& ...) [with T = {const char (&)[19], std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, curl_proxytype&}; std::string = std::__cxx11::basic_string<char>; format_string<T ...> = basic_format_string<char, const char (&)[19], std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, curl_proxytype&>]'
../xbmc/utils/log.h:129:31:   required from 'void CLog::FormatAndLogInternal(spdlog::level::level_enum, const std::string_view&, Args&& ...) [with Args = {const char (&)[19], std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, curl_proxytype&}; std::string_view = std::basic_string_view<char>]'
../xbmc/utils/log.h:100:39:   required from 'static void CLog::Log(spdlog::level::level_enum, const std::string_view&, Args&& ...) [with Args = {const char (&)[19], std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, curl_proxytype&}; std::string_view = std::basic_string_view<char>]'
../xbmc/utils/log.h:80:8:   required from 'static void CLog::Log(int, const std::string_view&, Args&& ...) [with Args = {const char (&)[19], std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, curl_proxytype&}; std::string_view = std::basic_string_view<char>]'
../xbmc/utils/log.h:92:8:   required from 'static void CLog::Log(int, uint32_t, const std::string_view&, Args&& ...) [with Args = {const char (&)[19], std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, curl_proxytype&}; uint32_t = unsigned int; std::string_view = std::basic_string_view<char>]'
../xbmc/filesystem/CurlFile.cpp:850:13:   required from here
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-12.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/fmt/core.h:1691:7: error: static assertion failed: Cannot format an argument. To make type T formattable provide a formatter<T> specialization: https://fmt.dev/latest/api.html#udt
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-12.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/fmt/core.h:1691:7: note: 'formattable' evaluates to false
```


## Possible Fix
<!--- Not obligatory, but suggest a fix or reason for the bug -->
<!--- Put your text below this line -->

- Previous version of this issue was in 8.1
  -  #20789
  - #21203

### To Reproduce
Steps to reproduce the behavior:
<!--- Provide a link to a live example, or an unambiguous set of steps to -->
<!--- reproduce this bug. Include code to reproduce, if relevant -->
<!--- Put your text below this line -->
1. 
2.
3.


### Debuglog
<!--- Put your text below this line -->
<!--- A debuglog is always mandatory when creating an issue. Provide one! -->
The debuglog can be found here:



### Screenshots 
Here are some links or screenshots to help explain the problem:
<!--- Put your text below this line -->



## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:
<!--- How has this bug affected you? What were you trying to accomplish? -->
<!--- Put your text below this line -->



### Your Environment
Used Operating system:
<!--- Include as many relevant details about the environment you experienced the bug in -->
<!--- Put your text below this line. Checkboxes can easily be ticked once issue is created -->
 - [ ] Android
 - [ ] iOS
 - [ ] tvOS
 - [X] Linux
 - [ ] macOS
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: LibreELEC 12 (Master)
 - Kodi version: kodi-21.0a2-Omega



<!--- End of this issue -->
*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

--- a/xbmc/guilib/GUIColorButtonControl.cpp
+++ b/xbmc/guilib/GUIColorButtonControl.cpp
@@ -188,7 +188,8 @@ void CGUIColorButtonControl::RenderInfoText()
 void CGUIColorButtonControl::ProcessInfoText(unsigned int currentTime)
 {
   CRect labelRenderRect = m_labelInfo.GetRenderRect();
-  bool changed = m_labelInfo.SetText(StringUtils::Format("#{:08X}", m_imgBoxColor));
+  bool changed = m_labelInfo.SetText(
+      StringUtils::Format("#{:08X}", static_cast<UTILS::COLOR::Color>(m_imgBoxColor)));
   // Set Label X position based on image mask control position
   float textWidth = m_labelInfo.GetTextWidth() + 2 * m_labelInfo.GetLabelInfo().offsetX;
   float textPosX = m_imgColorMask->GetXPosition() - textWidth;

--- a/xbmc/utils/log.h
+++ b/xbmc/utils/log.h
@@ -46,6 +46,10 @@ class dist_sink;
 } // namespace sinks
 } // namespace spdlog
 
+#if FMT_VERSION >= 100000
+using fmt::enums::format_as;
+#endif
+
 class CLog : public ISettingsHandler, public ISettingCallback
 {
 public:

--- a/xbmc/utils/log.h
+++ b/xbmc/utils/log.h
@@ -48,6 +48,14 @@ class dist_sink;
 
 #if FMT_VERSION >= 100000
 using fmt::enums::format_as;
+
+namespace fmt
+{
+template<typename T, typename Char>
+struct formatter<std::atomic<T>, Char> : formatter<T, Char>
+{
+};
+} // namespace fmt
 #endif
 
 class CLog : public ISettingsHandler, public ISettingCallback

--- a/xbmc/windows/GUIWindowHome.cpp
+++ b/xbmc/windows/GUIWindowHome.cpp
@@ -78,7 +78,8 @@ void CGUIWindowHome::Announce(ANNOUNCEMENT::AnnouncementFlag flag,
 {
   int ra_flag = 0;
 
-  CLog::Log(LOGDEBUG, LOGANNOUNCE, "GOT ANNOUNCEMENT, type: {}, from {}, message {}", flag, sender, message);
+  CLog::Log(LOGDEBUG, LOGANNOUNCE, "GOT ANNOUNCEMENT, type: {}, from {}, message {}",
+            AnnouncementFlagToString(flag), sender, message);
 
   // we are only interested in library changes
   if ((flag & (ANNOUNCEMENT::VideoLibrary | ANNOUNCEMENT::AudioLibrary)) == 0)

